tools/mo/openvino/tools/mo/moc_frontend/pipeline.py

```diff
@@ -213,4 +213,48 @@ def shape_to_array(shape: PartialShape):
             input_model.set_partial_shape(place, new_partial_shape)
 
     ngraph_function = moc_front_end.convert(input_model)
+    if argv.framework == "pytorch":
+        pytorch_process_after_convert(argv, ngraph_function)
+    return ngraph_function
+
+
+def pytorch_process_after_convert(argv, ngraph_function):
+    import torch
+    from openvino.frontend.pytorch.decoder import pt_to_ov_type_map
+    example_inputs = getattr(argv, "example_input", None)
+    input_signature = getattr(argv, "input_signature", None)
+    provide_shapes = argv.input_shape is not None
+    if example_inputs is not None:
+        inputs = [example_inputs] if isinstance(example_inputs, torch.Tensor) else example_inputs
+        if input_signature is not None and isinstance(inputs, dict):
+            ordered_inputs = []
+            upd_sign = []
+            for key in input_signature:
+                if key not in inputs:
+                    continue
+                ordered_inputs.append(inputs[key])
+                upd_sign.append(key)
+            inputs = ordered_inputs
+            input_signature = upd_sign
+        for idx, input_tensor in enumerate(ngraph_function.inputs):
+            if isinstance(inputs, (list, tuple)):
+                input_data = inputs[idx]
+            else:
+                input_data = list(inputs.values())[idx]
+            pt_dtype = input_data.dtype if isinstance(input_data, torch.Tensor) else type(input_data)
+            dtype = pt_to_ov_type_map.get(str(pt_dtype))
+            if dtype is None:
+                raise f"Unknown input dtype {pt_dtype}"
+
+            input_tensor.get_node().set_element_type(dtype)
+            if input_signature is not None:
+                tensor = input_tensor.get_tensor()
+                input_names = tensor.names
+                input_names.update(input_signature[idx])
+                tensor.set_names(input_names)
+            if not provide_shapes:
+                # prevent dynamic rank issue
+                shape = [-1] * len(input_data.shape)
+                input_tensor.get_node().set_partial_shape(PartialShape(shape))
+        ngraph_function.validate_nodes_and_infer_types() 
```

Comment:
again it is a duplicate of https://github.com/openvinotoolkit/openvino/blob/master/tools/mo/openvino/tools/mo/moc_frontend/pipeline.py.
We need to adopt PT case to use `moc_pipeline`

Reply:
it is done before running convert, in my case, it is required to do after, 
input_model.inputs() returns empty list and as I know right now, it is impossible to implement.
When, it will be ready, I remove this part of code

Reply:
Let me understand this first. Is it a correct assumption that we need this part to set dynamic dimensions in shapes and types when they do not match ones that provided with `example_input`? If that is correct, I don't understand why we need to set types in this case, because when using `example_input` we should use `torch.jit.trace` and all the types are fused into the model. No?

@eaidova, I don't understand why we cannot implement setting types and shapes in PT FE -- looks like it is just moving this code to PT FE in the right place? I mean why this is impossible to implement fe.inputs()? Do you mean that now it is not implemented and this is why we are implementing it here?

Reply:
yes, I mean that it is not implemented now. I have nothing against to migrate into common flow, when it will be implemented (I believe, it is separated task as well as support extensions, right?)

Reply:
Now we should understand that we miss to utilize some functionality (like freezing inputs) that MO can demonstrate if FE supports it or gracefully complains on it. In the common part different logic already processed gracefully by MO and we need to utilize this logic for PT FE.
I think this is the main task of this story to have appropriate integration of PT FE into MO.

Reply:
Yes we are missing all this things for now. Will be added later. This is a temporary workaround because a lack of shapes is a killer for the whole feature. There is the ticket particularly for inputs: CVS-102339. So this is not a long-term solution.